### DISCUSSION TITLE
How to get the latest VS MSbuild path in C# usig vswhere nuget

### DISCUSSION THREAD

**Discussion by Berkunath** [cake-build/cake#3109] [Q&A]
I can get the Latest VS installed details as below in cake. Can you please share the corresponding c# code for the below.

 var vsInstallation =
                VSWhereLatest(new VSWhereLatestSettings { Requires = "Microsoft.Component.MSBuild" })
                ?? VSWhereLatest(new VSWhereLatestSettings { Requires = "Microsoft.Component.MSBuild", IncludePrerelease = true });

Regards,
Berk

**Comment by devlead** [comment]
Could you clarify the question, your above example is C# code, are you trying to use outside Cake script or do you mean for how you use it with i.e. `MSBuild` alias within a Cake script?

**Reply by Berkunath** [reply]
Hi @devlead 

We have cake script and equivalent c# project that used MSBuild for compilation. While trying to compile WinUI Desktop projects, we are facing issue "Could not locate MSBuild". 

So we have get MSBuild ToolPath of the the latest VS using vswhere in cake script using below code and used Toolpath in MSBuild, it fixes the issue in build using script.

var vsInstallation =
VSWhereLatest(new VSWhereLatestSettings { Requires = "Microsoft.Component.MSBuild" })
?? VSWhereLatest(new VSWhereLatestSettings { Requires = "Microsoft.Component.MSBuild", IncludePrerelease = true });

I need to apply this workaround in equivalent c# project too.
